app/react/V2/Components/Forms/MultiSelect.tsx

```diff
@@ -59,6 +61,12 @@ const MultiSelect = ({
     onChange(newValue);
   };
 
+  useEffect(() => {
+    if (!isFirstRender) {
+      setCurrentValue(value);
```

Comment:
I'm not convinced that we should be manually calculating if it's a first render. Why do we want to set the value only if it's not the first render? The value we are setting comes from the props, and it's the value assigned to currentValue when the component mounts. So when it mounts we give it that value, and if it re renders we give it that same value again? Can a memoized value not be a better choice?

Reply:
For the record, we've seen that useEffect is required to update the state when the props have changed. The first render condition effectively is not needed.